Adding what's new entry for retentionConfig in mco

dockerymick opened the pull request:
https://github.com/open-cluster-management/backlog/issues/12628

dockerymick committed:
Adding what's new entry for retentionConfig in mco

dockerymick committed:
updating the obs API with descriptions

dockerymick commented:
@marcolan018 are there any properties that need to be added into the API file, based on this [doc](https://docs.google.com/document/d/1qejBPv2D1nx7JbsOXFSXOKfvnyirH0VR3vnhlJsXWz4/edit)?

dockerymick committed:
Update whats_new.adoc

dockerymick committed:
Update whats_new.adoc

marcolan018 commented:
@dockerymick 
there are more changes in the retention related configuration. It's under a new section named as "retentionConfig" under "spec", and some of them have the new default values like below

```
spec:
  retentionConfig:
    blockDuration: 2h
    cleanupInterval: 5m
    deleteDelay: 48h
    retentionInLocal: 24h
    retentionResolution1h: 0d
    retentionResolution5m: 180d
    retentionResolutionRaw: 30d
```

marcolan018 commented:
based on changes in https://github.com/open-cluster-management/multicluster-observability-operator/pull/553
the latest retention config should as below:

"spec", and some of them have the new default values like below

```
spec:
  advanced:
    retentionConfig:
      blockDuration: 2h
      cleanupInterval: 5m
      deleteDelay: 48h
      retentionInLocal: 24h
      retentionResolution1h: 0d
      retentionResolution5m: 180d
      retentionResolutionRaw: 30d
```

@clyang82 fyi

dockerymick committed:
Update observability_enable.adoc

updating the default values

marcolan018 commented on observability/observability_enable.adoc:
```diff
@@ -184,20 +184,21 @@ spec:
+      retentionResolution1h: 0d # How long to retain samples of 1 hour in bucket
+      retentionResolution5m: 180d # How long to retain samples of 5 minutes in bucket
+      retentionResolutionRaw: 30d # How long to retain raw samples in bucket
+    storageConfig: # Specifies the storage to be used by Observability
```
for "storageConfig", it's still under "spec"
only "retentionConfig" is moved to "spec.advanced"

dockerymick committed:
Update observability.json.adoc

Updated the YAML section of this API, will update the table with descriptions next

dockerymick commented on apis/observability.json.adoc:
```diff
@@ -104,11 +104,35 @@ __required__|Parameters describing the MultiClusterObservability resource to be
+        "retentionResolutionRaw":
+        }
+      }
+      "storageConfig": {
```
Should this be in the advance section or spec?

marcolan018 commented:
/lgtm
